### PR TITLE
Wallshoves, kick stuns and disarms v4

### PR DESCRIPTION
**Pull request by Rukofamicom**
## About The Pull Request
- Removes disarm from wallshove
- Duration of knockdown from wallshove is now 50% longer
- Shoving someone currently knocked down no longer stuns them, and instead forces them to drop their actively held item
- Removes the ability to "knock down" someone who is already in the knockdown state by shoving them into a wall. 
- Increases the action cooldown when performing disarms

## Why It's Good For The Game

You can call it a "skill issue" all you want, but Beestation is not supposed to be a skill based combat server. It is a newbie friendly roleplaying server where players are expected to act appropriately in the face of a threat. Unarmed combatants being able to turn the tide of battle to the extent that wallshoves currently enable is not conducive to this. Being unarmed should be significantly disadvantageous. 

This PR still rewards positional combat and also ensures that no player is ever totally helpless even if they are unarmed. A six second knockdown will slow an attacker substantially and is plenty of time to disengage from combat. 

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Disarming kick
image

Defending against the kick with grab intent
image

</details>

## Changelog
:cl:
del: Players who are shoved into walls no longer immediately drop their items, but are knocked down for 50% longer
add: Shoving someone who is knocked down will cause them to drop their held item(s)
del: Removed 4-second hardstun from shoving someone who is knocked down.
tweak: Gives disarm attacks an increased action cooldown
balance: Walls and empty hands are no longer the single most dangerous thing to worry about in combat.
/:cl:

### CODE REVIEW

**Comment by Wilchenx** [comment]
apparently newbie friendly equals hugbox

**Comment by DrDuckedGoose** [comment]
Completely removing the stun, in any measure, is partially erking but, I'd be interested to see how it feels.

File: code/modules/mob/living/carbon/human/species.dm [hunk 1]
```diff
@@ -1497,13 +1497,19 @@ GLOBAL_LIST_EMPTY(roundstart_races)
...
+				if(!(target.a_intent == INTENT_GRAB))
+					target.drop_all_held_items()
+					target.visible_message("<span class='danger'>[user.name] kicks \the [target_held_item] out of [target]'s hand!</span>",
+									"<span class='danger'>[user.name] kicks \the [target_held_item] out of your hand!</span>", null, COMBAT_MESSAGE_RANGE)
+					log_combat(user, target, "disarms [target_held_item]")
+					return
+				else
+					target.visible_message("<span class='danger'>[user.name] attempts to kick \the [target_held_item] away, but [target] maintains their grip!</span>",
+									"<span class='danger'>[user.name] fails to kick \the [target_held_item] out of your hand!</span>", null, COMBAT_MESSAGE_RANGE)
+					return
```
Comment (PowerfulBacon) [inline code/modules/mob/living/carbon/human/species.dm]: I don't think these should return and should instead continue, so you can push someone into a bin and make them drop items at the same time

Comment (Rukofamicom) [reply code/modules/mob/living/carbon/human/species.dm]: 👍

**Comment by Kratairo** [comment]
I am unsure of how making combat less skill reliant adds to RP (wouldnt it even make sense IC that a seasoned fighter would be better in combat?)

**Comment by FriendlyContractor** [comment]
> You can call it a "skill issue" all you want, but Beestation is not supposed to be a skill based combat server

???

Im sorry ruko but what you propose does not really fix the issue. What is going to happen with the changes you propose is that now instead of killing person you shoved with their own weapon(assuming they manage to put it away quickly enough) you will kill them with your improvised weapon(kitchen knife, bone saw, etc) and they can not fight back because if they pull out weapon of their own to defend themselves you can immediately disarm them and use their weapon against them while they are on the ground

**Comment by Patternseeker** [comment]
Trust the plan this one will be merged. Only two more PRs

**Comment by Rukofamicom** [comment]
> What is going to happen with the changes you propose is that now instead of killing person you shoved with their own weapon(assuming they manage to put it away quickly enough) you will kill them with your improvised weapon(kitchen knife, bone saw, etc) and they can not fight back because if they pull out weapon of their own to defend themselves you can immediately disarm them and use their weapon against them while they are on the ground

Yes, I 100% agree with you @FriendlyContractor. The PR won't actually change much now that it has been neutered to this point and I'm not 100% on even keeping it open since I was forced to re-add the easy disarm mechanic.

**Comment by PerishedFraud** [comment]
This one unironically seems more reasonable. It's still a crewkill PR - something we should be getting after antags are reworked, but I can't see this actually ruining rounds unlike prior iterations.

**Comment by zeskorion** [comment]
this isn't the *worst* disarm rework. better than what we have, certainly

File: code/modules/mob/living/carbon/human/species.dm [hunk 2]
```diff
@@ -1452,6 +1452,7 @@ GLOBAL_LIST_EMPTY(roundstart_races)
 	return
 
 /datum/species/proc/disarm(mob/living/carbon/user, mob/living/carbon/human/target, datum/martial_art/attacker_style)
+	user.changeNext_move(CLICK_CD_MELEE*2) //longer cooldown after disarm actions before next action can be taken.
```
Comment (PowerfulBacon) [inline code/modules/mob/living/carbon/human/species.dm]: This should be a seperate define

**Comment by PowerfulBacon** [comment]
I'm fine with this one, but its going on testmerge for some time first

File: code/modules/mob/living/carbon/human/species.dm (hunk 2 above)
Comment (Rukofamicom) [reply code/modules/mob/living/carbon/human/species.dm]: done.

**Comment by FriendlyContractor** [comment]
Im sorry but as i said this PR has been neutered to the point where it does absolutely nothing to change the situation. Like what is is even the point of merging in its current state

**Comment by Rukofamicom** [comment]
> Like what is is even the point of merging in its current state

It's not as far as I'd like to go, but it is still definitely better than the current state. Loss of the 4 second stun means you can still fight back to at least *some* degree even if you have to put your weapon away. Disarm having a doubled cooldown also means the player who was disarmed gets the first chance to pick their dropped weapon back up and spamming shoves for the knockdown is a bit harder.

**Comment by FriendlyContractor** [comment]
I did not think about it but you are right on the doubled cooldown. This might actually be better change than i thought

**Comment by Rukofamicom** [comment]
My opinion has completely changed after seeing it in action in the current round. This does exactly what I wanted pretty well. 

I don't actually even think it needs to go further, this is a good spot for it to be. You can't really use your weapon once you're knocked down if you get repeatedly disarmed, but it also isn't so easily used against you either.

**Comment by DatBoiTim** [comment]
> It's still a crewkill PR

You realize this effects everyone right?

Moreover, sometimes you're just gonna die. Being unkillable isn't conducive to RP. It removes a primary motivator for characters: the fear of death. It also entirely undermines people who want to play heroic characters, since there's no real stakes.

Death is gonna happen. Sometimes you will wind up in situations where you can't do anything about it. Players aren't meant to be one men armies. 

Antags should absolutely be a threat to the crew, because that what makes them drive the round. Having to think twice about charging someone is bringing you closer to actual RP. Now just base that decision on some simple personality traits about the character and boom you actually have roleplay.

Enough whining about this nerfing crew. They're average folks on a spacestation that has specially trained agents for a hostile entity aboard. Some people are going to die. Not everyone is gonna be capable of being a hero.

Death is also entirely meaningless since players are easily revivable under most circumstances. And if the body is beyond recovery you can often get cloned. And I am still in favor of making death more significant because there's no real reason to fear death.

There's no stakes to anything on the crews end but a lot more for antags. If you don't like dying go play animal crossing or something.

> something we should be getting after antags are reworked

Balancing antags is completely worthless if you can 1-click win them while unarmed.

**Comment by Isy232** [comment]
> I am unsure of how making combat less skill reliant adds to RP (wouldnt it even make sense IC that a seasoned fighter would be better in combat?)

You as a player may be very experienced in combat, but your character of a scientist or engineer is not a trained soldier.

If we do enough changes RP wise, some sort of background system could exist, but thats on a whole other point that we're way too far away from. Most engineers wouldn't have combat training so ingame none do.

> It also entirely undermines people who want to play heroic characters, since there's no real stakes.

I'm nitpicking but there still is barely any stakes with medical as it is, rather its just a bit harder to be a "hero".

I like these changes a lot in theorey, have yet to see too much ingame. The idea of making it harder to turn the tables on an antag and kill them with their own weapon (which was quite easy) is sensible and instead making it easier to run away and escape from the antag to compensate sounds great.

**Comment by DatBoiTim** [comment]
> I'm nitpicking but there still is barely any stakes with medical as it is, rather its just a bit harder to be a "hero".

I mean there should absolutely be more to medical to make heroism more life threatening.

It also shouldn't be easy as 1 click to become a hero.

The two can and should coexist.

**Comment by heepox** [comment]
Rip CQC
and kind of RIP all other Martial Arts too.
Rip Tabling.
Rip Non-lethal unarmed combat.

also now i can just charge at people with a weapon and no slips without a care in the world... so pog?

also rip crowd fighting?

If we really must go here then grabbing people who are knocked down should prevent them from breaking the tier 1 grip untill they get up.

**Comment by Rukofamicom** [comment]
> Rip CQC
and kind of RIP all other Martial Arts too.

Sleeping carp user murderboned the station with no issues while this was testmerged.
If there's some specific issue this is causing with martial arts, please let me know so I can resolve it. They seemed to work fine to me in general

> Rip Tabling.

Table interactions were not changed, except at most the ability to chain tableslams was which was shitty to allow in the first place anyway

> Rip Non-lethal unarmed combat.

If you're referring to being unarmed vs someone else who wants to be lethal and wasn't? Yeah that was kind of the entire point of this. 
If you're referring to combat where both parties don't have lethal intention, I have no idea what you're talking about.

**Review by PowerfulBacon — APPROVED** [review]

**Comment by PerishedFraud** [comment]
> You realize this effects everyone right?

Right. It will definitely affect the unarmed crew and the armed or otherwise advantaged antags equally.

> Moreover, sometimes you're just gonna die.

And with this sentence, you threw away all validity away from your post. 

If i just have to die, give all the antags instakills and get it over with.

People use "death is part of the game" as an excuse for poor balancing. They call out comparisons to more traditional games like mafia where combat doesn't even exist.  If we "just have to die" then what's the point of combat even being an option? Some false sense of hope or security? A boner for the antags for thinking their targets could've won? 

Please, never post that shit again. Thank you.

**Comment by Panicat** [comment]
This is stupid. What is the point of disarm then? Might as well remove intents and become TG... Do you know what disarm means? Even removing the stun and keeping disarm would be better than this...

**Pull request merged**